File: integration-tests/pkg/types/process.go
package types

type ProcessInfo struct {
	Name    string
	ExePath string
	Uid     int
	Gid     int
	Pid     int
	Args    string
}

type ProcessLineage struct {
	Name          string
	ExePath       string
	ParentUid     int
	ParentExePath string
}

File: collector-rs/crates/collector-types/src/process.rs
use crate::container::ContainerId;

#[derive(Debug, Clone)]
pub struct ProcessInfo {
    pub pid: u32,
    pub uid: u32,
    pub gid: u32,
    pub exe_path: String,
    pub args: String,
    pub container_id: ContainerId,
}

#[derive(Debug, Clone)]
pub struct LineageInfo {
    pub parent_uid: u32,
    pub parent_exe_path: String,
}
